WcaOnRails/app/controllers/registrations_controller.rb

```diff
@@ -192,6 +192,47 @@ def register
     end
   end
 
+  def process_payment
+    competition = competition_from_params
+    if current_user
+      registrations = competition.registrations
+      registration = registrations.find_by_user_id(current_user.id)
+    else
+      flash[:danger] = "Please register for the competition first, and then pay the entry fee."
+      redirect_to competition_register_path
+      return
+    end
+    token = params[:stripeToken]
+    Stripe.api_key = ENVied.STRIPE_API_KEY
+
+    begin
+      charge = Stripe::Charge.create({
+        amount: registration.outstanding_entry_fees.cents,
+        currency: registration.entry_fee_currency_code,
+        source: token,
+        description: "Registration payment",
+        metadata: registration.user.wca_id,
+        application_fee: 100,
```

Comment:
I probably should remove the application_fee for now, but this demonstrates how the WCA could take a fee for each registration. This would go straight into the WCA Stripe account.